Enable Support User

jyeshe opened the pull request:
## Description

This PR *[adds/changes/fixes]... (A description of your work goes here.)*

Closes #__

## Validation steps

1. *(How can a reviewer validate your work?)*

## Additional notes for the reviewer

1. *(Is there anything else the reviewer should know or look out for?)*

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Pre-submission checklist

- [ ] I have performed a **self-review** of my code.
- [ ] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [ ] I have updated the **changelog**.
- [ ] I have ticked a box in "AI usage" in this PR

jyeshe commented:
Unrelated flaky:
```
  1) test AI Assistant: users can sort chat sessions (LightningWeb.WorkflowLive.EditTest)
     test/lightning_web/live/workflow_live/edit_test.exs:3312
     Assertion with == failed
     code:  assert first_link |> Floki.attribute("id") == ["session-#{newer_session.id}"]
     left:  ["session-349f032c-331b-4370-b81a-fa6ee7aad440"]
     right: ["session-664e1d7e-f34e-4934-b1d3-2b4fd7cdbb9a"]
     stacktrace:
       test/lightning_web/live/workflow_live/edit_test.exs:3421: (test)
```

stuartc reviewed (changes requested):
This looks fantastic @jyeshe, thank you! Very thorough. I did leave a note about the `nil` context/subject argument for the ProjectUsers policy function. I also added some tests to the project users policy tests for the changes you introduced, one of them is failing (I'm assuming that we shouldn't allow support users to be able to do anything without context), please feel free to change the code or change the test if I'm not seeing this this right way - your thoughts?

jyeshe commented on lib/lightning_web/live/dashboard_live/user_projects_section.ex:
```diff
@@ -37,7 +37,10 @@ defmodule LightningWeb.DashboardLive.UserProjectsSection do
   defp switch_sort_direction("desc"), do: "asc"
 
   defp assign_projects(%{assigns: assigns} = socket) do
-    sort_param = "#{assigns.sort_key}_#{assigns.sort_direction}"
+    sort_param = {
+      String.to_existing_atom(assigns.sort_key),
+      String.to_existing_atom(assigns.sort_direction)
+    }
```
With a tuple I could use the separate values to sort the concat of two queries here:
```
{sort_key, sort_direction} = Keyword.get(opts, :order_by, {:name, :asc})

[user_projects, support_projects]
|> Enum.concat()
|> Enum.uniq_by(& &1.id)
|> Enum.sort_by(&Map.get(&1, sort_key), sort_direction)
```
I used an union select in first place first but there were bigger changes for supporting the latest_updated sorting due to different bindings on the two queries.

jyeshe commented on lib/lightning/policies/project_users.ex:
```diff
@@ -79,18 +80,53 @@ defmodule Lightning.Policies.ProjectUsers do
+        nil
+      )
+      when action in @project_user_actions,
+      do: support_user
```
It would be not for any project but only to  the ones you have access to. On LV we are fine but let me double-check the controllers.

jyeshe replied:
Hey @stuartc there was no controller using (the provision is handled by another Policies module). On the LV the `access_project` check before anything else, when the support user access is granted.

jyeshe commented:
> This looks fantastic @jyeshe, thank you! Very thorough. I did leave a note about the `nil` context/subject argument for the ProjectUsers policy function. I also added some tests to the project users policy tests for the changes you introduced, one of them is failing (I'm assuming that we shouldn't allow support users to be able to do anything without context), please feel free to change the code or change the test if I'm not seeing this this right way - your thoughts?

I see your concern. It was thinking about as well but the LV is hardened in a way that none of those actions will be reachable. Thanks for the additional test cases, I had added plenty and the only not covered before was the export on test coverage report. Let me check the ones you've added why they are not passing.

jyeshe commented:
> > This looks fantastic @jyeshe, thank you! Very thorough. I did leave a note about the `nil` context/subject argument for the ProjectUsers policy function. I also added some tests to the project users policy tests for the changes you introduced, one of them is failing (I'm assuming that we shouldn't allow support users to be able to do anything without context), please feel free to change the code or change the test if I'm not seeing this this right way - your thoughts?
> 
> I see your concern. I was thinking about as well but the LV is hardened in a way that none of those actions will be reachable (the nil is a semantics for project_users after checking that the user.support_user is `true`). Thanks for the additional test cases, I had added plenty and the only not covered before was the export on test coverage report. Let me check the ones you've added why they are not passing.

Done (passing compared to editor)
